scripts/host-check

```diff
@@ -1600,46 +1589,75 @@ def run_checks_all_platforms(extra_checks: List[str]) -> bool:
-
-        if base_success and platform_success:
-            platforms_supported.add(platform)
-        else:
+        plat_results = perform_checks(checks)
+        b_and_p_results = list(base_results.values()) + list(
+            plat_results.values()
+        )
+        if any(result.is_failure() for result in b_and_p_results):
             platforms_not_supported.add(platform)
+        elif any(result is CheckState.ERROR for result in b_and_p_results):
+            platforms_errored.add(platform)
+        else:
+            platforms_supported.add(platform)
 
     if extra_checks:
-        extras_supported, extras_not_supported = run_extra_checks(extra_checks)
-
-    print(f"\n{DOUBLE_DASHED_LINE}")
+        print("")  # insert a newline
+        print_heading("Extra checks")
+        for extra_check in sorted(extra_checks):
+            print_subheading(f"{extra_check} checks")
+            extra_results = perform_checks(EXTRA_CHECKS_MAP[extra_check])
+
+            if any(check.is_failure() for check in extra_results.values()):
+                extras_not_supported.add(extra_check)
+            elif any(c is CheckState.ERROR for c in extra_results.values()):
+                extras_errored.add(extra_check)
+            else:
+                extras_supported.add(extra_check)
 
+    # Print a summary of what is and is not supported
+    print_without_newline(f"\n{DOUBLE_DASHED_LINE}")
     if platforms_supported:
         print_without_newline(
-            "XR platforms supported: " + ", ".join(sorted(platforms_supported))
-        )
-    elif not platforms_supported:
-        print_without_newline(
-            "!! Host NOT set up correctly for any XR platforms !!"
+            "\nXR platforms supported: "
+            + ", ".join(sorted(platforms_supported))
         )
-    if platforms_not_supported and platforms_supported:
+    if platforms_not_supported:
         print_without_newline(
             "\nXR platforms NOT supported: "
             + ", ".join(sorted(platforms_not_supported))
         )
-
+    if platforms_errored:
+        print_without_newline(
+            "\n!! One or more platform checks could not be performed, see errors above !!"
+        )
+    # Print the result of any extra checks
     if extra_checks:
-        print_extra_checks_summary(extras_supported, extras_not_supported)
-
+        print_without_newline(f"\n{DASHED_LINE}")
+        if extras_supported:
+            print_without_newline(
+                "\nExtra checks passed: " + ", ".join(sorted(extras_supported))
+            )
+        if extras_not_supported:
+            print_without_newline(
+                "\nExtra checks failed: "
+                + ", ".join(sorted(extras_not_supported))
+            )
+        if any(c is CheckState.ERROR for c in extra_results.values()):
+            print_without_newline(
+                "\nExtra checks errored: " + ", ".join(sorted(extras_errored))
+            )
```

Comment:
Not sure why this was changed either as it's duplicated, can just pass in the three "support", "not supported" and "errored"?